src/main/java/io/jenkins/stapler/idea/jelly/NamespaceUtil.java

```diff
@@ -0,0 +1,51 @@
+package io.jenkins.stapler.idea.jelly;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiManager;
+import com.intellij.psi.search.FileTypeIndex;
+import com.intellij.psi.search.GlobalSearchScope;
+import com.intellij.psi.xml.XmlTag;
+import java.util.HashMap;
+import java.util.Map;
+import org.kohsuke.stapler.idea.language.JellyFileType;
+import org.kohsuke.stapler.idea.psi.JellyFile;
+
+public class NamespaceUtil {
+
+    /** Collects all namespaces (and prefixes) from .jelly files in the project */
+    public static Map<String, String> collectProjectNamespaces(Project project) {
```

Comment:
Need to check the performance of this and how often its called and if there's any caching

Reply:
this seems to be called on every key press although I didn't notice any delay... might be worth optimising if it doesn't cause other issues